resources/migrations/001_update_migrations.yaml

```diff
@@ -5799,6 +5798,34 @@ databaseChangeLog:
         - customChange:
             class: "metabase.db.custom_migrations.DeleteScanFieldValuesTriggerForDBThatTurnItOff"
 
+  - changeSet:
+      id: v49.2024-05-07T10:00:00
+      author: qnkhuat
+      comment: Set revision.most_recent = true to latest revision and false to others. A redo of v48.00-008 for mysql
+      preConditions:
+        - onFail: MARK_RAN
+        - dbms:
+            type: mysql,mariadb
+      changes:
+        - sql:
+            dbms: mysql,mariadb
+            sql: >-
+              UPDATE revision AS r
+              JOIN (
+                  SELECT r.id,
+                        ROW_NUMBER() OVER (PARTITION BY r.model, r.model_id ORDER BY r.timestamp DESC, r.id DESC) AS row_num
+                    FROM revision AS r
+                    JOIN (
+                          SELECT model, model_id
```

Comment:
Thanks @piranha for the suggestion about making this more efficient by only updating infected rows.

Reply:
Is the query plan smart enough to avoid calculating row numbers for the other models, or does that need more jiggling? That also seems like a nice optimization if possible.